gem/gem.py

```diff
@@ -875,11 +875,19 @@ def strides_of(shape):
 
 
 def decompose_variable_view(expression):
-    """Extract ComponentTensor + FlexiblyIndexed view onto a variable."""
-    if isinstance(expression, Variable):
+    """Extract information from a shaped node.
+       Decompose ComponentTensor + FlexiblyIndexed."""
+    if (isinstance(expression, Variable) or
+       isinstance(expression, Inverse) or
+       isinstance(expression, Solve)):
```

Comment:
`isinstance` accepts a tuple of types to check. So `isinstance(expression, (Inverse, Solve, Variable))` is probably clearer.